### PR TITLE
GPL-977 Added biosero page for creating or failing partial destination plates

### LINKED ISSUE #74: GPL-977 Biosero - Create partial destination plate pages [Heron]
**User story**
As a lab user I would like to be able to create a partial destination plate in the LIMS in case of robot failure or because I don’t want to wait to do another run. I would like to do this via the existing destination creation screen in Lighthouse-UI.

**Who are the primary contacts for this story**
Philip
Andrew

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] scanning of the destination barcode and a selection of automation_system (Beckman or Biosero)

**Dependencies**
This story is blocked by the following dependencies:
Biosero API endpoint for destination view lookup

**Additional context**
Modify the existing destination creation endpoint in Lighthouse-UI (and change the tab name in the UX to be generic). Let the user choose which system the plate was made on. If Beckman; the DART view to get the plate information, if Biosero; use their plate view API endpoint to fetch the information. Creation in Sequencescape should be identical for both.

Coupled with Lighthouse story [GPL-974](https://github.com/sanger/lighthouse/issues/325)

### PR DESCRIPTION
**Pull request by andrewsparkes**
Closes #74

### CODE REVIEW

File: test/modules/lighthouse_service.spec.js [hunk 1]
```diff
@@ -560,12 +560,149 @@ describe('lighthouse_service api', () => {
...
+      const result = await lighthouse.failDestinationPlateBeckman(form)
 
       const expected = { success: false, errors: ['There was an error'] }
 
       expect(mock).toHaveBeenCalledTimes(1)
       expect(result).toEqual(expected)
     })
   })
+
+  describe('#createDestinationPlateBiosero', () => {
```
Comment (pjvv) [inline test/modules/lighthouse_service.spec.js:572]: This file is getting long, it might be worth splitting these tests into the separate systems.

Comment (andrewsparkes) [reply test/modules/lighthouse_service.spec.js:572]: yes ok

**Review by pjvv — APPROVED** [review]

**Review by emrojo — DISMISSED** [review]
I dont know too much about vuejs to give a better feedback, but from what I got the idea seems ok

File: components/BioseroCherrypickForm.vue [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      },
+    }
+  },
+  computed: {
+    formInvalid() {
+      const commonValidation =
+        this.form.username.trim() === '' ||
+        this.form.barcode.trim() === ''
+      if (this.action === 'fail') {
+        return commonValidation || this.form.failureType === ''
```
[the remark this answers is not in the document]
Comment (andrewsparkes) [reply components/BioseroCherrypickForm.vue]: changed

**Pull request merged**